tests/integration/test_app.py

```diff
@@ -0,0 +1,42 @@
+# Standard Library
+from time import sleep
+
+# Third Party
+from requests import get, post
+
+
+def test_documents_endpoints():
+    """ Tests calling '/documents' endpoint with PDF file and then '/documents/{document_id}'
+        and finally '/documents/{document_id}/pages/{page_id}'"""
+
+    files = {'document': ('pages_num_3.pdf', open('tests/resources/pages_num_3.pdf', 'rb'), 'application/pdf')}
+
+    # upload the document via '/documents' endpoint
+    response = post('http://localhost:8000/documents/', files=files)
+    response_json = response.json()
+
+    assert response.status_code == 200
+    assert len(response_json) == 1
+    assert "id" in response_json
+
+    # save the document id
+    document_id = response_json["id"]
+
+    # sleep 5 secs to be sure that the document is processed
+    sleep(5)
```

Comment:
What if 5 seconds is not enough (sometimes)?

What if I would want to speed the test up?

Reply:
The test(s) deserves many more improvements, however if we would keep this kind of test then instead of just having synchronous sleep, there should asynchronous sleep in loop checking the document status periodically till it is done or we reach the max timeout.

Reply:
Yep, I'd do it the same way